server/implementation/src/main/java/io/smallrye/graphql/execution/datafetcher/helper/ReflectionHelper.java

```diff
@@ -38,6 +41,39 @@ public ReflectionHelper(Operation operation, EventEmitter eventEmitter) {
         this.method = lookupMethod(operationClass, operation);
     }
 
+    public <T> T invokePrivileged(Object... arguments) {
+        final ClassLoader tccl = Thread.currentThread().getContextClassLoader();
+        return invokePrivileged(tccl, arguments);
+    }
+
+    public <T> T invokePrivileged(final ClassLoader classLoader, Object... arguments) {
+
+        try {
+            return (T) AccessController
+                    .doPrivileged(new PrivilegedExceptionAction<Object>() {
+                        @Override
+                        public Object run() {
+                            ClassLoader originalTccl = Thread.currentThread()
+                                    .getContextClassLoader();
+                            Thread.currentThread().setContextClassLoader(classLoader);
+
+                            try {
+                                return invoke(arguments);
+                            } catch (Exception ex) {
+                                // TODO: Handle this better.
+                                throw new RuntimeException(ex);
+                            } finally {
+                                if (originalTccl != null) {
+                                    Thread.currentThread().setContextClassLoader(classLoader);
+                                }
+                            }
+                        }
+                    });
+        } catch (PrivilegedActionException e) {
+            throw new RuntimeException(e);
```

Comment:
```suggestion
            throw new RuntimeException(e.getCause());
```

In my experience the PrivilegedActionException is mostly useless for problem solving and it adds noise to the stacktrace/logs, but is necessary in order to wrap the real exception.  So, I'd suggest just using the original exception.